tests/e2e/runner.py

```diff
@@ -50,6 +50,7 @@ def load_workloads(root_directory):
     return workloads
 
 
+# TODO(gitbuda): Cleanup doesn't work properly if e2e test fails unexpectedly -> FIX.
```

Comment:
Move this to another PR -> @hal-eisen-MG this might be (high probability) a source of flakyness.